### DISCUSSION TITLE
Using a digital potentiometer as a fake indoor thermistor?

### DISCUSSION THREAD

**Discussion by abmantis** [raomin/ESPAltherma#412] [General]
From the service manual, using an indoor thermistor with modulation enabled allows the heatpump to be more efficient in choosing the best LWT, since it can lower the LWT as the room reaches the desired room temperature. 
Has anyone tried this?

I am not sure how the HT behaves when using the ON/OFF relay instead, so maybe the fake thermistor may not be worth it...

**Comment by GO74** [comment]
Hm, isn't the heatpump also modulating based on the temperature of the preflow when using ON/OFF?

**Reply by abmantis** [reply]
I would assume so, it should modulate based on the in/out water temperature, but I wonder if it would be more optimized when using a temperature sensor instead of just ON/OFF. According to the service manual, it seems to be a bit more intelligent and take the current sensor temperature as an input to decide the output water temperature. But I am not sure about it, tough.

**Reply by abmantis** [reply]
Apparently it does not module (at least, not as I would expect it to).
You can see that it turns off the heatpump from 13:07 to 13:28 because it reached the setpoint water temp, despite the "external thermostat" relay was still on. If it was doing proper modulation, it would still keep the pump running at a lower LW temperature, to prevent cycling. 

![image](https://github.com/raomin/ESPAltherma/assets/974569/d7a2db0e-cc31-4c6b-9e65-34b789e252fb)

**Reply by Erwiiinst** [reply]
Could it be that the inverter is already operating at minimum power?

**Reply by abmantis** [reply]
I don't think so, because it is able to have a setpoint lower than 40°

**Reply by abmantis** [reply]
It seems that it turned off because the setpoint got lower too fast (due to weather compensation). 

It has been running for a while now and it keeps a nice constant LWT, without cycling:
![image](https://github.com/raomin/ESPAltherma/assets/974569/dd9b69d2-1852-4709-8eeb-0421916c388b)

So the only advantage of modulation would be to have varying setpoints depending on indoor temperature (instead of relying only on outdoor weather comp.).

**Comment by bashers222** [comment]
You cant use both the ON/OFF relay and the internal thermistor

**Reply by abmantis** [reply]
What internal thermistor? The WLT one?

**Reply by bashers222** [reply]
the room thermistor built into the controller, or a seperate wired thermistor.
Either way, you cant use those alongside something else triggering the On/OFF function.
I've tried

**Reply by abmantis** [reply]
Ah yeah. My intention was to use the indoor wired thermistor (a digital one, not the one build into the controller) instead of the on/off relay.

**Comment by abmantis** [comment]
By the way, this is one of the points on the installer reference guide regarding modulation:

![image](https://github.com/raomin/ESPAltherma/assets/974569/7f94fd49-22dc-44f6-b696-5e5a9e7f47d8)

It works only when using `Unit temperature control` (menu 2.9) set to `Room thermostat`, and should work when using both the internal sensor of the user interface, or when using a indoor sensor connected to the heatpump (option `External sensor` (9.B.1) set to `Room`).
